core/storage/classroom/gae_models.py

```diff
@@ -55,6 +58,22 @@ class ClassroomModel(base_models.BaseModel):
     # prerequisite topic IDs as value.
     topic_id_to_prerequisite_topic_ids = datastore_services.JsonProperty(
         indexed=False, required=False)
+    # Whether this classroom is published or not.
+    # False if classroom is hidden, True if published.
+    is_published = datastore_services.BooleanProperty(
+        indexed=True, required=True, default=False)
+    # The thumbnail filename of the classroom.
+    thumbnail_filename = datastore_services.StringProperty(indexed=True)
+    # The thumbnail background color of the classroom.
+    thumbnail_bg_color = datastore_services.StringProperty(indexed=True)
+    # The thumbnail size in bytes of the classroom.
+    thumbnail_size_in_bytes = datastore_services.IntegerProperty(indexed=True)
+    # The banner filename of the classroom.
+    banner_filename = datastore_services.StringProperty(indexed=True)
+    # The banner background color of the classroom.
+    banner_bg_color = datastore_services.StringProperty(indexed=True)
+    # The banner size in bytes of the classroom.
+    banner_size_in_bytes = datastore_services.IntegerProperty(indexed=True)
```

Comment:
1. Maybe let's create a structure with three fields and store it as a json in storage?
2. I don't think we should index these fields

Reply:
Done.